### PR TITLE
Update DG with diagram

### PR DESCRIPTION
**Pull request by Eriksen2411**

### CODE REVIEW

**Review by lirc572 — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -233,25 +233,35 @@ Step 5. After the project gets updated, `Model#saveProjectsFolder` is called to
 
 ### Add Event to Project Command [Implemented in v1.2]
```
Comment (lirc572) [inline docs/DeveloperGuide.md]: `[Implemented in v1.2]` can be removed since other sections do not have this.

**Review by lirc572 — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (vevek) [reply docs/DeveloperGuide.md]: Thanks for pointing it out @lirc572 !

Comment (vevek) [inline docs/DeveloperGuide.md]: This title can change to

Add Event Feature.

Can help to change Rouchen's Update command to Update Feature. To be consistent.

Then your section on Add command please move it below View Projects Feature and above the Update Feature.

**Review by vevek — CHANGES_REQUESTED** [review]
Please see the comments

**Review by lirc572 — DISMISSED** [review]

**Review by vevek — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -181,7 +181,51 @@ Step5. Execution of this method will result in a call to `MainWindow#selectProje
     * Cons:
         * `MainWindow` and `CommandResult` are not closed to modification. A new instruction to change the UI might require the addition of fields to `CommandResult` (boolean fields for instructions and other fields for related data) as well as a new conditional statement in `MainWindow#execute` to handle the new instruction. This makes it relatively difficult to add new instructions.
 
-### Update Commands
```
Comment (vevek) [inline docs/DeveloperGuide.md:184]: Please help to change Rc ones too

**Review by vevek — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (lirc572) [reply docs/DeveloperGuide.md:184]: It is already changed! It's at [line 228](https://github.com/AY2021S2-CS2103T-T11-2/tp/pull/364/files#diff-1a95edf069a4136e9cb71bee758b0dc86996f6051f0d438ec2c424557de7160bR228)! :blush:

Comment (lirc572) [reply docs/DeveloperGuide.md:184]: But probably can make `Feature` singular?

Comment (lirc572) [reply docs/DeveloperGuide.md:184]: > But probably can make `Feature` singular?

But I can do this in a new PR. Can merge this I think.

Comment (vevek) [reply docs/DeveloperGuide.md:184]: @lirc572 okay!

**Review by vevek — DISMISSED** [review]
LGTM!

**Comment by vevek** [comment]
There's a conflict with Sam's previous PR! @Eriksen2411

**Review by vevek — APPROVED** [review]
LGTM!

**Pull request merged**